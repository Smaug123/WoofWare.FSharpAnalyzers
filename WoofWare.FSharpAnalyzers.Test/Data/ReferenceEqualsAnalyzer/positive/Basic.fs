module ReferenceEqualsUsage

open System

let testBasicReferenceEquals () =
    let x = obj ()
    let y = obj ()
    Object.ReferenceEquals (x, y)

let testInlinedReferenceEquals () = Object.ReferenceEquals (obj (), obj ())

let testWithStrings () =
    let s1 = "hello"
    let s2 = "world"
    Object.ReferenceEquals (s1, s2)

let testWithValueTypes () =
    let x = 42
    let y = 42
    // This silently does the wrong thing - always returns false for boxed value types
    Object.ReferenceEquals (x, y)

let testMixedTypes () =
    let s = "test"
    let n = 42
    // This allows comparing different types unsafely
    Object.ReferenceEquals (s, n)

let testNullComparison () =
    let x = obj ()
    Object.ReferenceEquals (x, Unchecked.defaultof<obj>)
