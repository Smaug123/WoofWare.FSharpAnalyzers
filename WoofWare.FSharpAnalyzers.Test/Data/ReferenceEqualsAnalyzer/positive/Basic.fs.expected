WOOF-REFEQUALS | Warning | (8,4--8,33) | Object.ReferenceEquals should be avoided. It silently does the wrong thing on value types, and lacks type safety - it's too easy to accidentally compare objects of different types. Use a type-safe wrapper like 'let referenceEquals<'a when 'a : not struct> (x : 'a) (y : 'a) : bool = ...' instead. Suppress with comment including text 'ReferenceEquals allowed'.
WOOF-REFEQUALS | Warning | (11,4--11,43) | Object.ReferenceEquals should be avoided. It silently does the wrong thing on value types, and lacks type safety - it's too easy to accidentally compare objects of different types. Use a type-safe wrapper like 'let referenceEquals<'a when 'a : not struct> (x : 'a) (y : 'a) : bool = ...' instead. Suppress with comment including text 'ReferenceEquals allowed'.
WOOF-REFEQUALS | Warning | (16,4--16,35) | Object.ReferenceEquals should be avoided. It silently does the wrong thing on value types, and lacks type safety - it's too easy to accidentally compare objects of different types. Use a type-safe wrapper like 'let referenceEquals<'a when 'a : not struct> (x : 'a) (y : 'a) : bool = ...' instead. Suppress with comment including text 'ReferenceEquals allowed'.
WOOF-REFEQUALS | Warning | (22,4--22,33) | Object.ReferenceEquals should be avoided. It silently does the wrong thing on value types, and lacks type safety - it's too easy to accidentally compare objects of different types. Use a type-safe wrapper like 'let referenceEquals<'a when 'a : not struct> (x : 'a) (y : 'a) : bool = ...' instead. Suppress with comment including text 'ReferenceEquals allowed'.
WOOF-REFEQUALS | Warning | (28,4--28,33) | Object.ReferenceEquals should be avoided. It silently does the wrong thing on value types, and lacks type safety - it's too easy to accidentally compare objects of different types. Use a type-safe wrapper like 'let referenceEquals<'a when 'a : not struct> (x : 'a) (y : 'a) : bool = ...' instead. Suppress with comment including text 'ReferenceEquals allowed'.
WOOF-REFEQUALS | Warning | (32,4--32,56) | Object.ReferenceEquals should be avoided. It silently does the wrong thing on value types, and lacks type safety - it's too easy to accidentally compare objects of different types. Use a type-safe wrapper like 'let referenceEquals<'a when 'a : not struct> (x : 'a) (y : 'a) : bool = ...' instead. Suppress with comment including text 'ReferenceEquals allowed'.
